django/core/checks/translation.py

```diff
@@ -59,3 +69,13 @@ def check_language_settings_consistent(app_configs, **kwargs):
     if settings.LANGUAGE_CODE not in available_tags:
         return [E004]
     return []
+
+
+@register(Tags.translation)
+def check_setting_locale_paths(app_configs, **kwargs):
+    for path in settings.LOCALE_PATHS:
+        if not isinstance(path, (str, tuple, list)):
+            return [E005]
+        if isinstance(path, (tuple, list)) and len(path) != 2:
+            return [E006]
+    return []
```

Comment:
Rather than falling immediately it would be better to accumulate messages for all incorrect entries and highlight the incorrect values in the message. Look at how `E003` is done for an example.

This allows all errors to be fixed at once instead of the annoying run-fix-run-fix-… cycle